homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,185 @@
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> SwitchEntity:
+    """Create Switch Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    is_on = await coordinator.async_get_pss_state(capability=capabilities[0])
+    room_id: str = device["location"].removeprefix("/location/")
+    room_name: str = coordinator.rooms[room_id]
+    livisi_switch = LivisiSwitch(
+        hass,
+        config_entry,
+        coordinator,
+        unique_id=device.get("id"),
+        manufacturer=device.get("manufacturer"),
+        product=device.get("product"),
+        serial_number=device.get("serialNumber"),
+        device_type=device.get("type"),
+        name=config_details.get("name"),
+        capability_id=capabilities[0],
+        is_on=is_on,
+        room=room_name,
+    )
+    return livisi_switch
+
+
+class LivisiSwitch(SwitchEntity, CoordinatorEntity):
+    """Represents the Livisi Switch."""
+
+    def __init__(
+        self,
+        hass,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id,
+        manufacturer,
+        product,
+        serial_number,
+        device_type,
+        name,
+        capability_id,
+        is_on,
+        room,
+        version=None,
+    ):
+        """Initialize the Livisi Switch."""
+        self.hass: HomeAssistant = hass
+        self.config_entry: ConfigEntry = config_entry
+        self._attr_unique_id = unique_id
+        self._manufacturer = manufacturer
+        self._product = product
+        self._serial_number = serial_number
+        self._attr_model = device_type
+        self._attr_name = name
+        self._state = None
+        self._capability_id = capability_id
+        self._attr_is_on = is_on
+        self._room = room
+        self._version = version
+        self.aio_livisi = coordinator.aiolivisi
+        if is_on is None:
+            self._attr_is_available = False
+        else:
+            self._attr_is_available = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self._attr_unique_id))},
+            manufacturer=self._manufacturer,
+            model=self._attr_model,
+            name=self._attr_name,
+            suggested_area=self._room,
+            sw_version=self._version,
+            via_device=(DOMAIN, str(self._attr_unique_id)),
+        )
+        super().__init__(coordinator)
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return the device state."""
+        return self._attr_is_on
+
+    @property
+    def available(self) -> bool:
+        """Return if switch is available."""
+        return self._attr_is_available
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        if self.is_on is True:
```

Comment:
Will there be a problem with the device if we set the same pss state again? We want to avoid guarding against setting a state in polling entities as the entity may have the wrong state and then we wouldn't be able to change state.

Reply:
It will not be a problem if we set the pss state again, the only inconvenience will be that we send more requests for the device. I will remove the guard of the pss state.